### DISCUSSION TITLE
Whether docker distribution binary version 2.8.2 supports ciphers?

### DISCUSSION THREAD

**Discussion by rahulbapumore** [distribution/distribution#4463] [Q&A]
We checked inside docker distribution code, we got to know defaultciphersuote list used by docker distribution as below - 

	// tls 1.2 ciphers
	TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256         
	TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256    
	TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384         
	TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384       
	TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256  
	TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 
 
	// TLS 1.3 cipher suites.
	TLS_AES_128_GCM_SHA256      
	TLS_AES_256_GCM_SHA384      
	TLS_CHACHA20_POLY1305_SHA256

I am connecting docker distribution using openssl command and using above ciphers to check whether it supports given cipher or not, but all above cipher passed and got response saying cipher is not supported.
**So we are confused now whether distribution binary uses cipher or not? could you please clarify?**

Below is one sample example and its output from our installed environment - 
```
openssl s_client -connect localhost:5000  -tls1_2 -cipher ECDHE-ECDSA-AES128-GCM-SHA256
CONNECTED(00000003)
140154879032320:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:ssl/record/ssl3_record.c:332:

no peer certificate available

No client certificate CA names sent

SSL handshake has read 5 bytes and written 142 bytes
Verification: OK

New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1726225539
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
```

**Comment by rahulbapumore** [comment]
we are not configuring any ciphers in config.yaml for distribution

**Comment by milosgajdos** [comment]
`v2.x` releases are no longer maintained. Please upgrade to the latest v3 release.

**Comment by rahulbapumore** [comment]
Hi @milosgajdos 
Yeah will do..
But for now , do you see whatever question asked make sense?
We are really getting response for all ciphers as they are not supported for docker distribution ...
So really ciphers are supported or not?

**Reply by milosgajdos** [reply]
I know very little about how `v2.x` works and have no intention in finding out. The source code is there so dig in.
